apps/web/src/pages/Pools/EditPoolPage/components/ScreeningQuestions.tsx

```diff
@@ -0,0 +1,326 @@
+import React from "react";
+import { IntlShape, useIntl } from "react-intl";
+import {
+  FormProvider,
+  useFieldArray,
+  useForm,
+  useWatch,
+} from "react-hook-form";
+
+import { TableOfContents, Well } from "@gc-digital-talent/ui";
+import { Locales, errorMessages } from "@gc-digital-talent/i18n";
+import {
+  Repeater,
+  TextArea,
+  Submit,
+  WordCounter,
+  countNumberOfWords,
+} from "@gc-digital-talent/forms";
+import { notEmpty } from "@gc-digital-talent/helpers";
+
+import {
+  CreateScreeningQuestionInput,
+  UpdateScreeningQuestionInput,
+  LocalizedString,
+  PoolAdvertisement,
+  Scalars,
+  AdvertisementStatus,
+  UpdatePoolAdvertisementInput,
+} from "~/api/generated";
+import { EditPoolSectionMetadata } from "~/types/pool";
+
+import { useEditPoolContext } from "./EditPoolContext";
+
+const TEXT_AREA_ROWS = 3;
+const TEXT_AREA_MAX_WORDS = 100;
+
+type ScreeningQuestionValue = {
+  id?: Scalars["ID"];
+  question: LocalizedString;
+};
+
+type FormValues = {
+  questions?: Array<ScreeningQuestionValue>;
+};
+
+export type ScreeningQuestionsSubmitData = Pick<
+  UpdatePoolAdvertisementInput,
+  "screeningQuestions"
+>;
+
+const WordLimit = ({ locale, index }: { locale: Locales; index: number }) => {
+  const value = useWatch({ name: "questions" });
+
+  return (
+    <div data-h2-margin="base(-x.5, 0, x1, 0)" data-h2-text-align="base(right)">
+      <WordCounter
+        text={value[index].question[locale] || ""}
+        wordLimit={TEXT_AREA_MAX_WORDS}
+      />
+    </div>
+  );
+};
+
+const maxWordValidator = (value: string, intl: IntlShape) => {
+  return (
+    countNumberOfWords(value) <= TEXT_AREA_MAX_WORDS ||
+    intl.formatMessage(errorMessages.overWordLimit, {
+      value: TEXT_AREA_MAX_WORDS,
+    })
+  );
+};
+
+interface ScreeningQuestionsProps {
+  poolAdvertisement: PoolAdvertisement;
+  sectionMetadata: EditPoolSectionMetadata;
+  onSave: (submitData: ScreeningQuestionsSubmitData) => void;
+}
+
+const ScreeningQuestions = ({
+  poolAdvertisement,
+  sectionMetadata,
+  onSave,
+}: ScreeningQuestionsProps) => {
+  const intl = useIntl();
+  const { isSubmitting } = useEditPoolContext();
+
+  const dataToFormValues = (initialData: PoolAdvertisement): FormValues => ({
+    questions:
+      initialData?.screeningQuestions
+        ?.filter(notEmpty)
+        .map(({ id, question }) => ({
+          id: id || "new",
+          question: {
+            en: question?.en || "",
+            fr: question?.fr || "",
+          },
+        })) || [],
+  });
+
+  const methods = useForm<FormValues>({
```

Comment:
When say editing a question you already have saved, "you have unsaved changes" appears to remind you to press "Save screening questions"
But when deleting, no reminder pops up to press "Save" to actually delete the question you want to delete
https://recordit.co/NROAjh7n7I

A similar reminder that the red trash can doesn't mean you actually deleted the question would be helpful, that you must press save afterwards.
Prevents having to come back to re-delete a question, a big pain if you were to delete, not save, and hit publish. And that if you accidentally pressed the red icon, it wasn't save yet.

Reply:
Good idea, added in [add removed items warning](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6303/commits/bd0024970c102b102e080b8a3a4c7e3658bdf840)